lib/recurly/api_errors.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * This file is automatically created by Recurly's OpenAPI generation process
+ * and thus any edits you make by hand will be lost. If you wish to make a
+ * change to this file, please create a Github issue explaining the changes you
+ * need and we will usher them to the appropriate places.
+ */
+'use strict'
+
+let ApiError = require('./ApiError')
+
+class BadRequestError extends ApiError { }
+
+class InternalServerError extends ApiError { }
+
+class ImmutableSubscriptionError extends ApiError { }
+
+class InvalidApiKeyError extends ApiError { }
+
+class InvalidApiVersionError extends ApiError { }
+
+class InvalidContentTypeError extends ApiError { }
+
+class InvalidPermissionsError extends ApiError { }
+
+class InvalidTokenError extends ApiError { }
+
+class NotFoundError extends ApiError { }
+
+class SimultaneousRequestError extends ApiError { }
+
+class TransactionError extends ApiError { }
+
+class UnauthorizedError extends ApiError { }
+
+class UnavailableInApiVersionError extends ApiError { }
+
+class UnknownApiVersionError extends ApiError { }
+
+class ValidationError extends ApiError { }
```

Comment:
The constructor of ApiError contains this line:
https://github.com/recurly/recurly-client-node/blob/4d3709a994519172627f67433c2be5a7fb4b202b/lib/recurly/ApiError.js#L4

I think it would be good to override this with the name of the specific class. Not sure if that work could be done in the ApiError class, but if it can, that would be the place to put it.

If node logs the stack trace, it looks something like this:
```
RecurlyApiError: Code can't be blank, Code is invalid
    at IncomingMessage.response.on (/Users/asuarez/code/v3/clients/node/lib/recurly/BaseClient.js:85:20)
    at IncomingMessage.emit (events.js:194:15)
    at endReadableNT (_stream_readable.js:1103:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```
So it might be nice to have something like:
```
RecurlyValidationError: Code can't be blank, Code is invalid
```
What do you think?

Reply:
Yeah, that should change. I'll investigate the best way to do it. Can probably just take the name of the class dynamically or maybe i should remove that line.

Reply:
If you remove that line, it probably won't add the `Recurly` part to the beginning (which I think is valuable). I would try to dynamically get the classname and prepend `Recurly` to it.